pkg/detectors/aws/aws.go

```diff
@@ -148,11 +184,15 @@ func (s scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 				for k, v := range extraData {
 					s1.ExtraData[k] = v
 				}
+				cacheItem.extra = s1.ExtraData
 				if verificationErr != nil {
 					s1.SetVerificationError(verificationErr, resSecretMatch)
+					cacheItem.verificationErr = verificationErr
 				}
 			}
 
+			// Cache the result.
+			s.credsCache.Set(rawV2, cacheItem)
```

Comment:
Should this be a pointer?

https://github.com/patrickmn/go-cache#usage (last example)

Reply:
Indeed, using a pointer is the right approach. I'm considering implementing a constructor to ensure that we consistently return a pointer. While this approach doesn't offer substantial protection since it's within the same package, it sets the groundwork for potential future refactoring. This could involve moving it into a separate package, from which it can be imported by all other detectors